### ISSUE TITLE
Improve adaptive mesh refinement metric

### ISSUE THREAD

**Issue by alexanderzimmerman** [geo-fluid-dynamics/phaseflow-fenics#111]
A problem is immediately evident in this example:

![image](https://user-images.githubusercontent.com/9367569/30385596-f5199d20-98a7-11e7-9ff6-2563487db264.png)

Currently whenever we refined, we're refining all cells that contain the PCI. In this example, the bottom part of the PCI has been refined so many more times than the top part because the top part is moving further into the domain. In danaila2014newton's method, they refine based on the enthalpy source term. If we use the time derivative term, then this would capture the cells where the PCI is actually moving. That should give a much better result.

**Comment by alexanderzimmerman** [comment]
Coming up with a better metric is quite difficult. I tried the change in the phase function over time; but then we lose the ability to resolve the PCI when it hasn't moved between cells.

For now I've addressed the issue in a much more simple manner with commit 30063731f61f0dc4f2b35f1555c1cf81ce717c08

Now we have an option for minimum cell diameter (i.e. circumradius).

The lack of a proper error estimator is still a big gap, so I'll open up another issue for that.
